### ISSUE TITLE
Incorrect RegExp in export-name rule leads to superfluously permissive

### ISSUE THREAD

**Issue by IllusionMH** [Microsoft/tslint-microsoft-contrib#632]
As was discovered in https://github.com/Microsoft/tslint-microsoft-contrib/pull/623#discussion_r230436742 RegExp in `export-name` rule has issue with improperly escaped `.` as well as missing check for line start/end. These problems leads to several problems.

For class name `SomeClass` it allows:
1. Wrong `.` escape: `SomeClassButAnyTextHere.tsx`, `SomeClassWithoutExtension`
1. No end of line check: `SomeClass.ts.or.not`
1. No start of line check: `LooksLikeSomeClass.ts`
1. Everything combined `NotSureThatSomeClassIsSomewhereInThis.long.filename`

According to description I would expect only `SomeClass.ts` or at least `SomeClass.suffix.tsx` (but better to make it as option).

What do you think about cases above and which should be disallowed by this rule?

**Comment by JoshuaKGoldberg** [comment]
Agreed, this is an existing bug that should be fixed as suggested; the author likely meant to escape the `.` properly.

**Comment by IllusionMH** [comment]
What about case 2 and 3?

**Comment by JoshuaKGoldberg** [comment]
> 2. No end of line check: `SomeClass.ts.or.not`

When would this case be typically hit? `SomeClass.spec.ts`, perhaps? My understanding of the RegExp's intent is that it was meant to allow (name of the exported thing) plus (any extension). It makes sense that we would consider `.spec.ts` to be the extension. Although the technical file extension is anything after the _last_ `.`, this rule's logic would be anything after the _first_ `.`.

> 3. No start of line check: `LooksLikeSomeClass.ts`

Yes, this looks like a bug.

**Comment by IllusionMH** [comment]
Actually with fixed check for start of the line there are 80 _(why not 93?)_ errors because of `export class Rule ...` in this repo 🤣 
Does in make sense to add this fix to 6.0? Might be breaking for someone.
Implementation and tests almost finished.
